cycler/__init__.py

```diff
@@ -223,31 +250,36 @@ def _from_iter(cls, label, itr):
         `Cycler`
             New 'base' cycler.
         """
-        ret = cls(None)
+        ret: Cycler[K, V] = cls(None)
         ret._left = list({label: v} for v in itr)
         ret._keys = {label}
         return ret
 
-    def __getitem__(self, key):
+    def __getitem__(self, key: slice) -> Cycler[K, V]:
         # TODO : maybe add numpy style fancy slicing
         if isinstance(key, slice):
             trans = self.by_key()
             return reduce(add, (_cycler(k, v[key]) for k, v in trans.items()))
         else:
             raise ValueError("Can only use slices with Cycler.__getitem__")
 
-    def __iter__(self):
-        if self._right is None:
-            for left in self._left:
-                yield dict(left)
+    def __iter__(self) -> Generator[dict[K, V], None, None]:
+        if self._right is None or self._left is None:
+            if self._left is not None:
+                for left in self._left:
+                    yield dict(left)
         else:
+            if self._op is None:
```

Comment:
mypy was not happy at certain points that op was potentially `None`... I suspect that since I made `op` just `Any`, mypy would not actually complain here anymore, but this and the extra check for the trivial cycler with `_left` and `_right` both `None` were added for mypy reasons